Makefile

```diff
@@ -8,53 +8,83 @@ else
 	GULP_ENVIRONMENT := production
 endif
 
-run_all: requirements npm_install frontend_build run_app
+.PHONY: run-all
+run-all: requirements npm-install frontend-build run-app
 
-run_app: show_environment virtualenv
+.PHONY: run-app
+run-app: show-environment virtualenv
 	${VIRTUALENV_ROOT}/bin/python application.py runserver
 
+.PHONY: virtualenv
 virtualenv:
 	[ -z $$VIRTUAL_ENV ] && [ ! -d venv ] && virtualenv venv || true
 
-upgrade_pip: virtualenv
+.PHONY: upgrade-pip
+upgrade-pip: virtualenv
 	${VIRTUALENV_ROOT}/bin/pip install --upgrade pip
 
-requirements: virtualenv upgrade_pip requirements.txt
+.PHONY: requirements
+requirements: virtualenv test-requirements upgrade-pip requirements.txt
 	${VIRTUALENV_ROOT}/bin/pip install -r requirements.txt
 
-requirements_for_test: virtualenv upgrade_pip requirements_for_test.txt
-	${VIRTUALENV_ROOT}/bin/pip install -r requirements_for_test.txt
-
-npm_install: package.json
+.PHONY: requirements-dev
+requirements-dev: virtualenv test-requirements requirements-dev.txt
+	${VIRTUALENV_ROOT}/bin/pip install -r requirements-dev.txt
+
+.PHONY: freeze-requirements
+freeze-requirements:
+	rm -rf venv-freeze
+	virtualenv -p python3 venv-freeze
+	$$(pwd)/venv-freeze/bin/pip install -r requirements-app.txt
+	echo '# This file is autogenerated. Do not edit it manually.' > requirements.txt
+	cat requirements-app.txt >> requirements.txt
+	echo '' >> requirements.txt
+	$$(pwd)/venv-freeze/bin/pip freeze -r requirements-app.txt | sed -n '/The following requirements were added by pip freeze/,$$p' >> requirements.txt
+	rm -rf venv-freeze
+
+.PHONY: npm-install
+npm-install: package.json
 	npm install
 
-frontend_build:
+.PHONY: frontend-build
+frontend-build:
 	npm run --silent frontend-build:${GULP_ENVIRONMENT}
 
-test: show_environment frontend_build test_pep8 test_python test_javascript
+.PHONY: test
+test: show-environment test-requirements frontend-build test-pep8 test-python test-javascript
```

Comment:
We kept them as separate steps so far. We could make an argument about `make test` running pip install each time now that the versions are pinned, but it feels like a separate conversation to me.